bittensor/_axon/__init__.py

```diff
@@ -441,19 +453,21 @@ def check_signature(
             message = f"{nonce}{sender_hotkey}{receptor_uuid}"
         else:
             raise Exception("Invalid signature version")
-        # Build the key which uniquely identifies the endpoint that has signed
-        # the message.
-        endpoint_key = f"{sender_hotkey}:{receptor_uuid}"
 
         if endpoint_key in self.nonces.keys():
             previous_nonce = self.nonces[endpoint_key]
             # Nonces must be strictly monotonic over time.
-            if nonce <= previous_nonce:
-                raise Exception("Nonce is too small")
+            lower_bound = clock.monotonic_ns() - bittensor.__blocktime__*1e9 # nonce should not be too much earlier then block_time
```

Comment:
The reference point of the monotonic clock functions (both `monotonic` and `monotonic_ns`) is undefined - the monotonic clocks are not global, such that nonce bounds cannot be established in this way. Generally, monotonic clocks are implemented based on system uptime timestamps, which obviously are not global.

You can easily test this by simply checking the return value on `monotonic_ns` on two different machines - you will see that the values are wildly different, and as such the bounds computed on the server cannot be used to bound the values generated by the clients.

Time boxing nonces is not useful nor necessary. If the staleness of the request is really a concern, another timestamp should be added to the signature composition, but it won't significantly improve the security of the signature.